src/NServiceBus.Core/Licensing/LicenseExpiredForm.cs

```diff
@@ -80,19 +84,62 @@ void browseButton_Click(object sender, EventArgs e)
 
         void PurchaseButton_Click(object sender, EventArgs e)
         {
-            Process.Start("http://particular.net/licensing");
+            Process.Start("https://particular.net/licensing");
         }
 
         void getTrialLicenseButton_Click(object sender, EventArgs e)
         {
-            if (CurrentLicense != null && CurrentLicense.IsTrialLicense)
+            var parameterCollection = new NameValueCollection
             {
-                Process.Start("http://particular.net/extend-your-trial-14");
+                {"NugetUser", IsNugetUser().ToString()},
+                {"PlatformInstaller", HasUserInstalledPlatform().ToString()},
+                {"TrialStartDate", TrialStartDateStore.GetTrialStartDate().ToString()}
+            };
+
+            UriBuilder builder;
+            if (CurrentLicense != null && !CurrentLicense.IsExtendedTrial)
+            {
+                // Original 14 day trial expired, give the user a chance to extend trial
+                builder = new UriBuilder("https://particular.net/extend-nservicebus-trial")
+                {
+                    Query = string.Join("&", parameterCollection.AllKeys.Select(key => $"{HttpUtility.UrlEncode(key)}={HttpUtility.UrlEncode(parameterCollection[key])}"))
+                };
             }
             else
             {
-                Process.Start("http://particular.net/extend-your-trial-45");
+                // Extended trial license expired, ask the user to Contact Sales
+                builder = new UriBuilder("https://particular.net/extend-your-trial-45")
+                {
+                    Query = string.Join("&", parameterCollection.AllKeys.Select(key => $"{HttpUtility.UrlEncode(key)}={HttpUtility.UrlEncode(parameterCollection[key])}"))
+                };
+            }
+
+            // Open the url with the querystrings
+            Process.Start(builder.Uri.AbsoluteUri);
+        }
+
+        bool IsNugetUser()
```

Comment:
@indualagarsamy Also the key being opened is in CurrentUser, not LocalMachine,  It would be extremely unusual for ACLs to be present here that would block it from being opened for read